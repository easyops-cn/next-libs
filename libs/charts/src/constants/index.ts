export * from "./format";
export * from "./data";
export * from "./datetimeRange";
export * from "./panel";
