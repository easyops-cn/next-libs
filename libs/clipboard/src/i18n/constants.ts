export const NS_CLIPBOARD = "clipboard";

export enum K {
  COPY_SUCCESS = "COPY_SUCCESS",
  COPY_FAILED = "COPY_FAILED",
}

export type Locale = { [key in K]: string };
