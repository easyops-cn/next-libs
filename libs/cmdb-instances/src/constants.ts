export const UploadFileUrl =
  "api/gateway/object_store.object_store.PutObject/api/v1/objectStore/bucket/cmdb-bucket/object";

export const DownloadFileUrl =
  "api/gateway/object_store.object_store.GetObject/api/v1/objectStore/bucket/cmdb-bucket/object";
