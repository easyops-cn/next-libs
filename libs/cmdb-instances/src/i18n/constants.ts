export const NS_LIBS_CMDB_INSTANCES = "libs-cmdb-instances";

export enum K {
  VALIDATE_MESSAGE_REQUIRED = "VALIDATE_MESSAGE_REQUIRED",
  DYNAMIC_FILTER = "DYNAMIC_FILTER",
  TIP = "TIP",
  DELETE = "DELETE",
  INVALID_OR_FORBIDDEN_IPS = "INVALID_OR_FORBIDDEN_IPS",
  INVALID = "INVALID",
  SELECT_FROM_CMDB = "SELECT_FROM_CMDB",
  IP_PLACEHOLDER = "IP_PLACEHOLDER",
  CLICK_TO_SELECT = "CLICK_TO_SELECT",
  MATCHING_REGULAR = "MATCHING_REGULAR",
  ADD = "ADD",
  ENTER_MULTIPLE_STRING_WITH_ENTER_KEY_AS_THE_SEPARATOR = "ENTER_MULTIPLE_STRING_WITH_ENTER_KEY_AS_THE_SEPARATOR",
  UNIQUE_NO_REPEAT = "UNIQUE_NO_REPEAT",
  TYPE_NO_SUPPORT_EDIT = "TYPE_NO_SUPPORT_EDIT",
  TEMPORARILY_NOT_CHOOSE = "TEMPORARILY_NOT_CHOOSE",
  ENUM_ERROR_TIP = "ENUM_ERROR_TIP",
  STRUCT_ERROR_TIP = "STRUCT_ERROR_TIP",
  LINK = "LINK",
  LINK_PLACEHOLDER = "LINK_PLACEHOLDER",
  TITLE = "TITLE",
  TITLE_PLACEHOLDER = "TITLE_PLACEHOLDER",
  CLICK_TO_FILTER = "CLICK_TO_FILTER",
  CLICK_TO_CANCEL_FILTER = "CLICK_TO_CANCEL_FILTER",
  CLICK_TO_HIDDEN = "CLICK_TO_HIDDEN",
  CLICK_TO_SHOW = "CLICK_TO_SHOW",
  RELATED_TO_ME = "RELATED_TO_ME",
  DISPLAY_OMITTED_INFORMATION = "DISPLAY_OMITTED_INFORMATION",
  CLEAR = "CLEAR",
  NORMAL_HOST = "NORMAL_HOST",
  CURRENT_FILTER_REQUIREMENTS = "CURRENT_FILTER_REQUIREMENTS",
  ADVANCED_SEARCH = "ADVANCED_SEARCH",
  CHOSEN_OPTIONS = "CHOSEN_OPTIONS",
  FUZZY_SEARCH = "FUZZY_SEARCH",
  BASIC_INFORMATION = "BASIC_INFORMATION",
  MORE = "MORE",
  DEFAULT_ATTRIBUTE = "DEFAULT_ATTRIBUTE",
  CANCEL = "CANCEL",
  CLOSE = "CLOSE",
  DISPLAY_SETTINGS = "DISPLAY_SETTINGS",
  FIELD_SETTINGS = "FIELD_SETTINGS",
  SEARCH_BY_FIELD_NAME = "SEARCH_BY_FIELD_NAME",
  SELECT_ALL = "SELECT_ALL",
  RESTORE_DEFAULT = "RESTORE_DEFAULT",
  SET_DEFAULT_DISPLAY_SUCCESS = "SET_DEFAULT_DISPLAY_SUCCESS",
  DEFAULT_DISPLAY = "DEFAULT_DISPLAY",
  CONFIRM = "CONFIRM",
  ONLY_ONE_INSTANCE_TO_ALLOWED = "ONLY_ONE_INSTANCE_TO_ALLOWED",
  OPERATOR_CONTAIN_DEFINE = "OPERATOR_CONTAIN_DEFINE",
  OPERATOR_NOT_CONTAIN_DEFINE = "OPERATOR_NOT_CONTAIN_DEFINE",
  OPERATOR_EQUAL_DEFINE = "OPERATOR_EQUAL_DEFINE",
  OPERATOR_NOT_EQUAL_DEFINE = "OPERATOR_NOT_EQUAL_DEFINE",
  OPERATOR_IS_EMPTY_DEFINE = "OPERATOR_IS_EMPTY_DEFINE",
  OPERATOR_IS_NOT_EMPTY_DEFINE = "OPERATOR_IS_NOT_EMPTY_DEFINE",
  OPERATOR_BETWEEN_DEFINE_TEXT = "OPERATOR_BETWEEN_DEFINE_TEXT",
  OPERATOR_BETWEEN_DEFINE_TEXT_TIME = "OPERATOR_BETWEEN_DEFINE_TEXT_TIME",
  OPERATOR_BETWEEN_DEFINE_TEXT_INT = "OPERATOR_BETWEEN_DEFINE_TEXT_INT",
  OPERATOR_BETWEEN_DEFINE_TEXT_THE = "OPERATOR_BETWEEN_DEFINE_TEXT_THE",
  SEARCH = "SEARCH",
  RESET = "RESET",
  CREATOR = "CREATOR",
  CREATION_TIME = "CREATION_TIME",
  MODIFIER = "MODIFIER",
  LAST_MODIFICATION_TIME = "LAST_MODIFICATION_TIME",
  DEVELOPMENT = "DEVELOPMENT",
  VIEW = "VIEW",
  ITEM_ADD_IN_BATCHES = "ITEM_ADD_IN_BATCHES",
  SAVE = "SAVE",
  SAVE_AND_CONTINUE = "SAVE_AND_CONTINUE",
  MODIFICATION = "MODIFICATION",
  NOT_MEET_REGEX = "NOT_MEET_REGEX",
  NOT_MEET_REGEX_DETAIL = "NOT_MEET_REGEX_DETAIL",
  NOT_MEET_JSON = "NOT_MEET_JSON",
  NOT_MEET_XML = "NOT_MEET_XML",
  CREATE_ANOTHER = "CREATE_ANOTHER",
  DELETE_STRUCT_CONFIRM_MSG = "DELETE_STRUCT_CONFIRM_MSG",
  DELETE_INSTANCE_CONFIRM_MSG = "DELETE_INSTANCE_CONFIRM_MSG",
  OPERATION = "OPERATION",
  VIEW_ALL_SELECTED_INSTANCES = "VIEW_ALL_SELECTED_INSTANCES",
  CHOOSE_INSTANCE = "CHOOSE_INSTANCE",
  CHOOSE_INSTANCE_NAME = "CHOOSE_INSTANCE_NAME",
  VIEW_ALL_DATA = "VIEW_ALL_DATA",
  FILTER_FROM_CMDB = "FILTER_FROM_CMDB",
  PAGINATION_TOTAL_TEXT = "PAGINATION_TOTAL_TEXT",
  PAGINATION_TOTAL_UNIT = "PAGINATION_TOTAL_UNIT",
  JUMP_TO = "JUMP_TO",
  INSTANCE_DETAIL = "INSTANCE_DETAIL",
  COPY = "COPY",
  COPY_SUCCESS = "COPY_SUCCESS",
  // FILTER_FROM_CMDB = "FILTER_FROM_CMDB",
  USERS = "USERS",
  USER_GROUPS = "USER_GROUPS",
  SWITCH = "SWITCH",
  USERS_RESULT_LABEL = "USERS_RESULT_LABEL",
  NO_DATA = "NO_DATA",
  USER_GROUPS_RESULT_LABEL = "USER_GROUPS_RESULT_LABEL",
  PERMISSION_WHITELIST = "PERMISSION_WHITELIST",
  FILTER_FROM_CMDB_TYPE = "FILTER_FROM_CMDB_TYPE",
  ALL_CLUSTER = "ALL_CLUSTER",
  FREE_SELECTION = "FREE_SELECTION",
  APP_SELECTION = "APP_SELECTION",
  ATTRIBUTE_NAME_REQUIRED = "ATTRIBUTE_NAME_REQUIRED",
  INSTANCE_SOURCE = "INSTANCE_SOURCE",
  INSTANCE_SOURCE_TOOLTIP = "INSTANCE_SOURCE_TOOLTIP",
  INSTANCE_SOURCE_TAG_TEXT = "INSTANCE_SOURCE_TAG_TEXT",
  COPY_SELECTED_IP = "COPY_SELECTED_IP",
  SELECT_COPY_DATA = "SELECT_COPY_DATA",
  ADVANCE_SEARCH_SINGLE_INPUT_PLACEHOLDER = "ADVANCE_SEARCH_SINGLE_INPUT_PLACEHOLDER",
  NUMBER_INPUT_PLACEHOLDER = "NUMBER_INPUT_PLACEHOLDER",
  FIX_HEADER = "FIX_HEADER",
  CANCEL_FIX_HEADER = "CANCEL_FIX_HEADER",
  VIEW_SPECIFIC_INSTANCES = "VIEW_SPECIFIC_INSTANCES",
  VIEW_MORE = "VIEW_MORE",
  CREATE_AND_BIND = "CREATE_AND_BIND",
  PERMISSION_DENIED = "PERMISSION_DENIED",
  CLICK_TO_FILTER_INSTANCE_SOURCE = "CLICK_TO_FILTER_INSTANCE_SOURCE",
  SEARCH_MODEL = "SEARCH_MODEL",
  UPLOAD_SIZE_LARGE_TEXT = "UPLOAD_SIZE_LARGE_TEXT",
  UPLOAD = "UPLOAD",
  BACKGROUND_SEARCH = "BACKGROUND_SEARCH",
  BACK = "BACK",
  RELATION_INSTANCE_TOOLTIP = "RELATION_INSTANCE_TOOLTIP",
}

export type Locale = { [key in K]: string };
