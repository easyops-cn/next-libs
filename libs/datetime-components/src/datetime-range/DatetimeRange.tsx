import React from "react";
import { Popover, Button, Radio, DatePicker, Icon } from "antd";
import { find, get } from "lodash";
import { RadioChangeEvent } from "antd/lib/radio";
import { TooltipPlacement } from "antd/lib/tooltip";
import moment from "moment";
import { RangePickerValue } from "antd/lib/date-picker/interface";

export type DateRangeType = "dateRange";
export const DATE_RANGE = "dateRange";
export type SpecifiedDateType = "specifiedDate";
export const SPECIFIED_DATE = "specifiedDate";

export interface SpecifiedDateValue {
  from: number;
  to: number;
}

export interface DateRange {
  type: DateRangeType;
  value: string;
}
export interface SpecifiedDateRange {
  type: SpecifiedDateType;
  value: SpecifiedDateValue;
}

export type RangeType = "default" | "custom";

export interface DatatimeRangeProps {
  initDateRange?: DateRange | SpecifiedDateRange;
  onConfirm?(dateRange: DateRange | SpecifiedDateRange): void;
  format?: string;
  type?: RangeType;
  customTimeRange?: RangeText[];
  placement?: TooltipPlacement;
}

export interface DatatimeRangeState {
  dateRange: DateRange | SpecifiedDateRange;
  type: SpecifiedDateType | DateRangeType;
  range: string | null;
  specifiedDate: RangePickerValue | null;
  visible: boolean;
  format: string;
}

export interface RangeText {
  range: string;
  text: string;
}

export const defaultRangeOptionList: RangeText[] = [
  {
    range: "now-1h",
    text: "近1小时"
  },
  {
    range: "now-1d",
    text: "近24小时"
  },
  {
    range: "now/d",
    text: "今天"
  },
  {
    range: "now-7d",
    text: "近7天"
  },
  {
    range: "now-30d",
    text: "近30天"
  }
];

export class DatetimeRange extends React.Component<
  DatatimeRangeProps,
  DatatimeRangeState
> {
  static defaultProps = {
    type: "default",
    customTimeRange: [] as RangeText[]
  };

  rangeOptionList: RangeText[];
  constructor(props: DatatimeRangeProps) {
    super(props);

    const defaultInitRange = {
      type: DATE_RANGE,
      value: "now-7d"
    };

    const customInitRange = {
      type: DATE_RANGE,
      value: get(this.props.customTimeRange, "[0].range")
    };
    const initDateRange: DateRange | SpecifiedDateRange =
      this.props.initDateRange ||
      ((this.props.type === "default"
        ? defaultInitRange
        : customInitRange) as DateRange);

    this.rangeOptionList =
      this.props.type === "default"
        ? defaultRangeOptionList
        : this.props.customTimeRange;
    this.state = {
      dateRange: initDateRange,
      type: initDateRange.type,
      range: initDateRange.type === DATE_RANGE ? initDateRange.value : null,
      specifiedDate:
        initDateRange.type === SPECIFIED_DATE
          ? [moment(initDateRange.value.from), moment(initDateRange.value.to)]
          : null,
      visible: false,
      format: this.props.format || "YYYY-MM-DD HH:mm:ss"
    };
  }

  getButtonText() {
    if (this.state.dateRange.type === DATE_RANGE) {
      return find(this.rangeOptionList, ["range", this.state.dateRange.value])
        .text;
    } else {
      return (
        moment(this.state.dateRange.value.from).format(this.state.format) +
        " - " +
        moment(this.state.dateRange.value.to).format(this.state.format)
      );
    }
  }

  handleRangeChange = (e: RadioChangeEvent) => {
    const dateRange: DateRange = {
      type: DATE_RANGE,
      value: e.target.value
    };
    this.setState({
      dateRange,
      type: DATE_RANGE,
      range: e.target.value,
      specifiedDate: null
    });
    if (this.props.onConfirm) {
      this.props.onConfirm(dateRange);
    }
    this.hide();
  };

  handleVisibleChange = (visible: boolean) => {
    this.setState({ visible });
  };

  hide = () => {
    this.setState({
      visible: false
    });
  };

  save = () => {
    let dateRange: DateRange | SpecifiedDateRange;
    if (this.state.type === DATE_RANGE) {
      dateRange = {
        type: this.state.type,
        value: this.state.range
      };
    } else {
      dateRange = {
        type: this.state.type,
        value: {
          from: +this.state.specifiedDate[0],
          to: +this.state.specifiedDate[1]
        }
      };
    }
    this.setState({ dateRange });
    this.props.onConfirm?.(dateRange);
    this.hide();
  };

  onDateChange = (v: RangePickerValue) => {
    this.setState({
      type: SPECIFIED_DATE,
      range: null,
      specifiedDate: v
    });
  };

  render(): React.ReactNode {
    const labelStyle = {
      display: "block",
      color: "#262626",
      lineHeight: "24px",
      fontSize: "16px",
      fontWeight: 500,
      marginBottom: "10px"
    };
    const radioGroupStyle = {
      marginBottom: "20px"
    };
    const datePickerStyle = {
      marginBottom: "30px"
    };
    const contentStyle = {
      margin: "16px"
    };
    const content = (
      <div style={contentStyle}>
        <div style={labelStyle}>快速选择：</div>
        <Radio.Group
          value={this.state.range}
          onChange={this.handleRangeChange}
          style={radioGroupStyle}
          className="btnGroup"
        >
          {this.rangeOptionList.map(item => (
            <Radio.Button
              value={item.range}
              key={item.range}
              className={item.range}
            >
              {item.text}
            </Radio.Button>
          ))}
        </Radio.Group>
        <div style={labelStyle}>指定范围：</div>
        <div style={datePickerStyle}>
          <DatePicker.RangePicker
            showTime={{
              defaultValue: [
                moment(this.state.format),
                moment(this.state.format)
              ]
            }}
            format={this.state.format}
            style={{ width: "380px" }}
            placeholder={["开始日期", "结束日期"]}
            value={this.state.specifiedDate}
            onChange={this.onDateChange}
            allowClear={false}
          />
        </div>
        <Button type="primary" onClick={this.save}>
          确定
        </Button>
      </div>
    );
    return (
      <Popover
        placement={this.props.placement ?? "bottom"}
        content={content}
        trigger="click"
        visible={this.state.visible}
        onVisibleChange={this.handleVisibleChange}
      >
        <Button>
          <Icon type="clock-circle" style={{ verticalAlign: "middle" }} />{" "}
          {this.getButtonText()}
        </Button>
      </Popover>
    );
  }
}
