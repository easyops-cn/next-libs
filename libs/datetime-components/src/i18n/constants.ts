export const NS_LIBS_DATETIME_COMPONENTS = "libs-datetime-components";

export enum K {
  LAST_AN_HOUR = "LAST_AN_HOUR",
  LAST_24_HOURS = "LAST_24_HOURS",
  TODAY = "TODAY",
  LAST_7_DAYS = "LAST_7_DAYS",
  LAST_30_DAYS = "LAST_30_DAYS",
  CONFIRM = "CONFIRM",
  SPECIFIED_RANGE = "SPECIFIED_RANGE",
  QUICK_SELECTION = "QUICK_SELECTION",
  START_DATE = "START_DATE",
  END_DATE = "END_DATE",
}

export type Locale = { [key in K]: string };
