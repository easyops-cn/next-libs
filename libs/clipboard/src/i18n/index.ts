import i18next from "i18next";
import { NS_CLIPBOARD } from "./constants";
import en from "./locales/en";
import zh from "./locales/zh";

export function addResourceBundle() {
  i18next.addResourceBundle("en", NS_CLIPBOARD, en);
  i18next.addResourceBundle("zh", NS_CLIPBOARD, zh);
}

addResourceBundle();
