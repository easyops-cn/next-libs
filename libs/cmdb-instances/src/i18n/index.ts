import i18next from "i18next";
import { NS_LIBS_CMDB_INSTANCES } from "./constants";
import en from "./locales/en";
import zh from "./locales/zh";

// istanbul ignore next
export function addResourceBundle(): void {
  i18next.addResourceBundle("en", NS_LIBS_CMDB_INSTANCES, en);
  i18next.addResourceBundle("zh", NS_LIBS_CMDB_INSTANCES, zh);
}
