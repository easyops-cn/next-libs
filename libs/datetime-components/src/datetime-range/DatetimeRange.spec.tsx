import React from "react";
import { shallow, mount } from "enzyme";
import { DatetimeRange, DATE_RANGE, SPECIFIED_DATE } from "./DatetimeRange";
import moment from "moment";
import i18n from "i18next";
import { NS_LIBS_DATETIME_COMPONENTS, K } from "../i18n/constants";

describe("DatetimeRange", () => {
  const wrapper = shallow(<DatetimeRange />);
  const component = wrapper.instance() as DatetimeRange;

  it("should work", () => {
    expect(wrapper).toBeTruthy();
  });
  it("test getButtonText now-7d", () => {
    expect(component.getButtonText()).toBe(
      i18n.t(`${NS_LIBS_DATETIME_COMPONENTS}:${K.LAST_7_DAYS}`)
    );
  });
  it("test getButtonText now-30d", () => {
    const dateRange = {
      type: DATE_RANGE,
      value: "now-30d",
    };
    wrapper.setState({ dateRange });
    expect(component.getButtonText()).toBe(
      i18n.t(`${NS_LIBS_DATETIME_COMPONENTS}:${K.NEARLY_30_DAYS}`)
    );
  });
  it("test getButtonText SPECIFIED_DATE", () => {
    const dateRange = {
      type: SPECIFIED_DATE,
      value: {
        from: 1556703515644,
        to: 1556789915644,
      },
    };
    wrapper.setState({ dateRange });
    expect(component.getButtonText()).toBe(
      "2019-05-01 17:38:35 - 2019-05-02 17:38:35"
    );
  });
  it("test onDateChange", () => {
    component.onDateChange([moment(1556703515644), moment(1556789915644)]);
    expect(wrapper.state("type")).toBe(SPECIFIED_DATE);
    expect(wrapper.state("range")).toBe(null);
    expect(wrapper.state("specifiedDate")).toEqual([
      moment(1556703515644),
      moment(1556789915644),
    ]);
  });

  it("test disabledDate", () => {
    expect(component.disabledDate(moment(1556703515645))).toBe(false);
  });

  it("test handleVisibleChange", () => {
    component.handleVisibleChange(true);
    expect(wrapper.state("visible")).toBe(true);
  });
});

describe("DatetimeRange selectNearDays", () => {
  const wrapper = shallow(<DatetimeRange selectNearDays={90} />);
  const component = wrapper.instance() as DatetimeRange;

  it("test disabledDate", () => {
    expect(component.disabledDate(moment().subtract(100, "days"))).toBe(true);
    expect(component.disabledDate(moment().subtract(10, "days"))).toBe(false);
    expect(component.disabledDate(moment().add(10, "days"))).toBe(true);
  });
});

describe("DatetimeRange rangeDays", () => {
  const wrapper = shallow(<DatetimeRange rangeDays={90} />);
  const component = wrapper.instance() as DatetimeRange;

  it("test disabledDate", () => {
    wrapper.setState({
      dates: [moment(), moment()],
    });
    expect(component.disabledDate(moment().add(100, "days"))).toBe(true);
    expect(component.disabledDate(moment().add(10, "days"))).toBe(false);
  });
});

describe("DatetimeRange onConfirm", () => {
  const onConfirm = jest.fn();
  const wrapper = shallow(<DatetimeRange onConfirm={onConfirm} />);
  const component = wrapper.instance() as DatetimeRange;
  it("test save SpecifiedDateRange", () => {
    const dateRange = {
      type: SPECIFIED_DATE,
      value: {
        from: 1556703515644,
        to: 1556789915644,
      },
    };
    wrapper.setState({
      type: SPECIFIED_DATE,
      specifiedDate: [moment(1556703515644), moment(1556789915644)],
    });
    component.save();
    expect(wrapper.state("dateRange")).toEqual(dateRange);
    expect(onConfirm).toHaveBeenCalled();
  });
  it("test save dateRange", () => {
    const dateRange = {
      type: DATE_RANGE,
      value: "now-30d",
    };
    wrapper.setState({
      type: DATE_RANGE,
      range: "now-30d",
    });
    component.save();
    expect(wrapper.state("dateRange")).toEqual(dateRange);
    expect(onConfirm).toHaveBeenCalled();
  });

  it("render custom time range", () => {
    const wrapper = mount(
      <DatetimeRange
        type="custom"
        customTimeRange={[
          {
            range: "now-30d",
            text: "近30天",
          },
          {
            range: "now-1y",
            text: "近一年",
          },
        ]}
      />
    );

    wrapper.find(".ant-btn").simulate("click");
    wrapper.update();

    expect(wrapper.find("Radio").at(0).text()).toEqual("近30天");
    expect(wrapper.find("Radio").at(1).text()).toEqual("近一年");
  });
});
