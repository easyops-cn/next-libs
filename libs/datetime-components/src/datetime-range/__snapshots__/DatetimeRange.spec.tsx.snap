// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatetimeRange should work 1`] = `
<Popover
  content={
    <div
      style={
        Object {
          "margin": "16px",
        }
      }
    >
      <div
        style={
          Object {
            "color": "#262626",
            "display": "block",
            "fontSize": "16px",
            "fontWeight": 500,
            "lineHeight": "24px",
            "marginBottom": "10px",
          }
        }
      >
        快速选择：
      </div>
      <RadioGroup
        buttonStyle="solid"
        className="btnGroup"
        onChange={[Function]}
        style={
          Object {
            "marginBottom": "20px",
          }
        }
        value="now-7d"
      >
        <RadioButton
          className="now-1h"
          value="now-1h"
        >
          近1小时
        </RadioButton>
        <RadioButton
          className="now-1d"
          value="now-1d"
        >
          近24小时
        </RadioButton>
        <RadioButton
          className="now/d"
          value="now/d"
        >
          今天
        </RadioButton>
        <RadioButton
          className="now-7d"
          value="now-7d"
        >
          近7天
        </RadioButton>
        <RadioButton
          className="now-30d"
          value="now-30d"
        >
          近30天
        </RadioButton>
      </RadioGroup>
      <div
        style={
          Object {
            "color": "#262626",
            "display": "block",
            "fontSize": "16px",
            "fontWeight": 500,
            "lineHeight": "24px",
            "marginBottom": "10px",
          }
        }
      >
        指定范围：
      </div>
      <div
        style={
          Object {
            "marginBottom": "30px",
          }
        }
      >
        <PickerWrapper
          allowClear={false}
          format="YYYY-MM-DD HH:mm:ss"
          locale={Object {}}
          onChange={[Function]}
          onOk={[Function]}
          onOpenChange={[Function]}
          placeholder={
            Array [
              "开始日期",
              "结束日期",
            ]
          }
          popupStyle={Object {}}
          showTime={
            Object {
              "defaultValue": Array [
                null,
                null,
              ],
            }
          }
          style={
            Object {
              "width": "380px",
            }
          }
          transitionName="slide-up"
          value={null}
        />
      </div>
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        确定
      </Button>
    </div>
  }
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  onVisibleChange={[Function]}
  overlayStyle={Object {}}
  placement="bottom"
  transitionName="zoom-big"
  trigger="click"
  visible={false}
>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
  >
    <Icon
      style={
        Object {
          "verticalAlign": "middle",
        }
      }
      type="clock-circle"
    />
     
    近7天
  </Button>
</Popover>
`;
